### PR TITLE
Dembed2 Domain

### PR DESCRIPTION
**Pull request by cywteow**
Added a resolver for Dembed2 Domain

### CODE REVIEW

File: script.module.resolveurl/lib/resolveurl/plugins/dembed.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import re
+import requests
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/dembed.py:20]: ResolveURL doesnt include requests module.

File: script.module.resolveurl/lib/resolveurl/plugins/dembed.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    name = "dembed2"
+    domains = ['dembed2.com']
+    pattern = r'(?://|\.)(dembed2\.com)/(?:streaming\.php|embedplus)\?id=([a-zA-Z0-9-]+)'
+    key = '93422192433952489752342908585752'
+    iv = '9262859232435825'
+
+    def get_media_url(self, host, media_id):
+        encryptedParams = self._getEncryptedParams(f'https://{host}/streaming.php?id={media_id}')
+        web_url = self.get_url(host, encryptedParams)
+        response = requests.get(web_url, headers={"X-Requested-With":"XMLHttpRequest"})
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/dembed.py:46]: need to be done with the net module

File: script.module.resolveurl/lib/resolveurl/plugins/dembed.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+        if response.status_code == 200:
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/dembed.py:48]: same as above

File: script.module.resolveurl/lib/resolveurl/plugins/dembed.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            sources = []
+            encryptedResult = response.json()
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/dembed.py:50]: jason.loads as requests module not available

File: script.module.resolveurl/lib/resolveurl/plugins/dembed.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+from resolveurl import common
+from resolveurl.resolver import ResolveUrl, ResolverError
+from resolveurl.plugins.__resolve_generic__ import ResolveGeneric
+from base64 import b64decode, b64encode
+from resolveurl.lib.pyaes import AESModeOfOperationCBC, Encrypter, Decrypter
+# from Cryptodome.Cipher import AES
+# from Cryptodome.Util.Padding import unpad
+# from Cryptodome.Util.Padding import pad
+from base64 import b64decode, b64encode
+from urllib.parse import urlsplit, parse_qs, urlparse
```
Comment (Gujal00) [inline script.module.resolveurl/lib/resolveurl/plugins/dembed.py:31]: use six.moves as this will break Kodi <19 support currently in ResolveURL

**Review by Gujal00 — CHANGES_REQUESTED** [review]
Thanks mate, there are a couple of issues with this PR as ResolveURL doesn't include requests library and also the use of urllib.parse which makes PY3 only.

If you want I can redo the plugin based on your code in a compliant way. Thanks

**Comment by cywteow** [comment]
> Thanks mate, there are a couple of issues with this PR as ResolveURL doesn't include requests library and also the use of urllib.parse which makes PY3 only.
> 
> If you want I can redo the plugin based on your code in a compliant way. Thanks

Can you make the changes for me as Python is not my main language. Thank you for your time.

**Comment by Gujal00** [comment]
Closing this PR as the code has been rewritten to work on py2 and 3 in https://github.com/Gujal00/ResolveURL/commit/605094bf2d389fb0625f58d3be40c451af0f3518

**Comment by shellc0de27** [comment]
@Gujal00 Just a thought.....
This is a vidcloud9 domain can just be added to `vidcloud9.py` without even modifying headers as you can see below.
`SMR Link Tester: Link Supported: |https://dembed2.com/streaming.php?id=MzY3NjUy&title=American+Pickers+-+Season+23+Episode+17+-+Knucklehead+Blue&typesub=SUB&sub=&cover=Y292ZXIvYW1lcmljYW4tcGlja2Vycy1zZWFzb24tMjMucG5n| Resolvers: VidCloud9
SMR Link Tester: Link Resolved: |https://dembed2.com/streaming.php?id=MzY3NjUy&title=American+Pickers+-+Season+23+Episode+17+-+Knucklehead+Blue&typesub=SUB&sub=&cover=Y292ZXIvYW1lcmljYW4tcGlja2Vycy1zZWFzb24tMjMucG5n|https://hls3x.vidfiles.net/videos/hls/njCuYAEM5HGEEbpb1TSBsw/1661225521/367652/951a17ecbd942e39fa18883f9c1cfcee/ep.17.v0.1661130646.m3u8|User-Agent=Mozilla%2F5.0+%28Windows+NT+10.0%3B+Win64%3B+x64%3B+rv%3A68.0%29+Gecko%2F20100101+Firefox%2F68.0&Referer=https%3A%2F%2Fmembed.net%2Floadserver.php%3Fid%3DMzY3NjUy|`

**Comment by Gujal00** [comment]
@shellc0de27 i did check it, but as the key and iv are different and even the resolved file is different, I added the new one
```INFO <general>: SMR Link Tester: Playing Link: |https://dembed2.com/streaming.php?id=MzM4MjU0&title=Immortal+Samsara+%282022%29+episode+15&typesub=SUB|
INFO <general>: SMR Link Tester: Link Supported: |https://dembed2.com/streaming.php?id=MzM4MjU0&title=Immortal+Samsara+%282022%29+episode+15&typesub=SUB| Resolvers: dembed2
INFO <general>: SMR Link Tester: Link Resolved: |https://dembed2.com/streaming.php?id=MzM4MjU0&title=Immortal+Samsara+%282022%29+episode+15&typesub=SUB|https://cache.387e6278d8e06083d813358762e0ac63.com/222835338184.m3u8|User-Agent=Mozilla%2F5.0+%28Windows+NT+10.0%3B+Win64%3B+x64%3B+rv%3A68.0%29+Gecko%2F20100101+Firefox%2F68.0|```
Try that link above in browser and see what it plays and try with vidcloud9 resolver and see what it plays

**Comment by shellc0de27** [comment]
Oh snap completely different. wow....

**Comment by Gujal00** [comment]
@shellc0de27 yep looks like dembed2 site logic is just a clone of membed, but the actual media_ids and streams are different :)

**Comment by shellc0de27** [comment]
@Gujal00 Awesome catch bro :)

**Comment by shellc0de27** [comment]
Thanks for also clarifying. Was thinking to save time when they change and only updating once vs twice. But yeah their different :P

**Pull request closed without merging**